accepted/0000-dependency-selector-syntax.md

```diff
@@ -0,0 +1,421 @@
+# `npm query` Command & Dependency Selector Syntax
+
+## Summary
+
+Introduce a new `npm query` commmand which exposes a new dependency selector syntax (informed by & respecting many aspects of the [CSS Selectors 4 Spec](https://dev.w3.org/csswg/selectors4/#relational)).
+
+## Motivation
+
+- Standardize the shape of, & querying of, dependency graphs with a robust object model, metadata & selector syntax
+- Leverage existing, known language syntax & operators from CSS to make disperate package information broadly accessible
+- Unlock the ability to anwser complex, multi-faceted questions about dependencies, their relationships & associative metadata
+- Consolidate redundant logic of similar query commands in `npm` (ex. `npm fund`, `npm ls`, `npm outdated`, `npm audit` ...)
```

Comment:
I think this is an extremely good goal to have. One thing I'd likely end up recommending is that the method to do this is exposed via an easy-to-use API to lower the barrier for others _outsite_ of the core npm CLI maintainers to be able to easily add commands via the query language in the future.